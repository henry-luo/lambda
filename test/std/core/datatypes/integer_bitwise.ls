// Test: Integer Bitwise Operations
// Category: core/datatypes
// Type: positive
// Expected: 1

// Test bitwise operations
let a = 0b1100  // 12
let b = 0b1010  // 10

// Bitwise AND
let and = a & b  // 0b1000 (8)

// Bitwise OR
let or = a | b   // 0b1110 (14)

// Bitwise XOR
let xor = a ^ b   // 0b0110 (6)

// Bitwise NOT
let not_a = ~a    // Depends on integer size

// Left shift
let left_shift = a << 2  // 0b110000 (48)

// Right shift
let right_shift = a >> 1  // 0b0110 (6)

// Unsigned right shift (logical shift)
let unsigned_right = a >>> 1  // 0b0110 (6) for positive numbers

// Test with negative numbers
let neg = -1
let neg_shift = neg >> 1  // Arithmetic shift (sign-extended)
let neg_ushift = neg >>> 1 // Logical shift (zero-extended)

// Final check
1
