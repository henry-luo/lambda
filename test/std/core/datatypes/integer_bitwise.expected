09:15:42 [NOTICE] Start transpiling test/std/core/datatypes/integer_bitwise.ls...
09:15:42 [ERROR] Syntax tree has errors.
09:15:42 [ERROR] Syntax error at Ln 7, Col 9 - 7, Col 10: ERROR
09:15:42 [ERROR] ERROR node at Ln 7, Col 9 - 7, Col 10
09:15:42 [ERROR] Syntax error at Ln 8, Col 9 - 8, Col 10: ERROR
09:15:42 [ERROR] ERROR node at Ln 8, Col 9 - 8, Col 10
09:15:42 [ERROR] Syntax error at Ln 20, Col 13 - 20, Col 14: ERROR
09:15:42 [ERROR] ERROR node at Ln 20, Col 13 - 20, Col 14
09:15:42 [ERROR] Syntax error at Ln 20, Col 13 - 20, Col 14: ERROR
09:15:42 [ERROR] ERROR node at Ln 20, Col 13 - 20, Col 14
09:15:42 [ERROR] Syntax error at Ln 23, Col 21 - 23, Col 22: ERROR
09:15:42 [ERROR] ERROR node at Ln 23, Col 21 - 23, Col 22
09:15:42 [ERROR] Syntax error at Ln 26, Col 22 - 26, Col 23: ERROR
09:15:42 [ERROR] ERROR node at Ln 26, Col 22 - 26, Col 23
09:15:42 [ERROR] Syntax error at Ln 29, Col 25 - 29, Col 27: ERROR
09:15:42 [ERROR] ERROR node at Ln 29, Col 25 - 29, Col 27
09:15:42 [ERROR] Syntax error at Ln 33, Col 22 - 33, Col 23: ERROR
09:15:42 [ERROR] ERROR node at Ln 33, Col 22 - 33, Col 23
09:15:42 [ERROR] Syntax error at Ln 34, Col 23 - 34, Col 25: ERROR
09:15:42 [ERROR] ERROR node at Ln 34, Col 23 - 34, Col 25
09:15:42 [ERROR] Error: Failed to compile the function.
##### Script 'test/std/core/datatypes/integer_bitwise.ls' executed: #####
null
